### PR TITLE
Implement RFC 7405.

### PR DESCRIPTION
**Pull request by timothee-haudebourg**
Hi, this PR is to implement case sensitive string introduced by [RFC 7405](https://www.rfc-editor.org/rfc/rfc7405).

### CODE REVIEW

**Comment by duesee** [comment]
Hey @timothee-haudebourg, this looks all very good. Thanks! I will try to properly review this PR in the following few days. Don't mind the failing license check in CI for now.

File: Cargo.toml [hunk 1]
```diff
@@ -18,3 +18,4 @@ abnf-core = "0.5.0"
 quickcheck        = "1"
 quickcheck_macros = "1"
 rand              = "0.8"
+rand_distr        = "0.4"
```
Comment (duesee) [inline Cargo.toml]: Add newline at end of file.

File: src/lib.rs [hunk 2]
```diff
@@ -631,6 +696,12 @@ mod tests {
         println!("{:?}", res);
     }
 
+    #[test]
+    fn test_char_val() {
+        let char_val = std::fs::read_to_string("examples/assets/char_val.abnf").unwrap();
```
Comment (duesee) [inline src/lib.rs]: Can we "inline" that from the file?

File: examples/assets/char_val.abnf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+a = "foo"
+b = %i"bar"
+c = %s"baz"
+e = %I"BAR"
+f = %S"BAZ"
```
Comment (duesee) [inline examples/assets/char_val.abnf]: Can we "inline" that into the test? (I.e., remove this file.)

File: src/lib.rs [hunk 4]
```diff
@@ -301,12 +301,44 @@ fn option<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str, Node,
...
+    ))(input)
+}
+
+/// ```abnf
+/// case-insensitive-string = [ "%i" ] quoted-string
+/// ```
+fn case_insensitive_string<'a, E: ParseError<&'a str>>(
+    input: &'a str,
+) -> IResult<&'a str, &str, E> {
+    let marker = preceded(char('%'), alt((char('i'), char('I'))));
```
Comment (duesee) [inline src/lib.rs:324]: Hehe, seems you cared about everything! :-) I would not have added support for "%I" and "%S". But... it's what the RFC tells us to do, so ... :+1:

**Review by duesee — CHANGES_REQUESTED** [review]
Thanks, again! I have only two minor nits.

I suspect that you have a usecase for RFC 7405 support and will try to release a new version to crates.io later after fixing the license check in CI.

**Comment by timothee-haudebourg** [comment]
@duesee Thanks! Yes we have a use case for it there spruceid/treeldr/pull/35, did you figure out the CI problem? :smile:

**Comment by duesee** [comment]
Oh, thanks for the reminder! :-) https://github.com/duesee/abnf/pull/22 should fix the failing CI. Could you address the remaining two comments? This will be good to go then 👍🏻

File: src/lib.rs (hunk 2 above)
Comment (timothee-haudebourg) [reply src/lib.rs]: I hope the way I inlined it is ok.

**Comment by timothee-haudebourg** [comment]
@duesee sorry for the delay. Did you have time to see the changes?

**Comment by duesee** [comment]
No worries. Thanks again for the PR and for taking care of the requested changes! This should be good to go (when CI is happy) :+1:

**Review by duesee — APPROVED** [review]
Changes approved.

**Comment by duesee** [comment]
I'll update the version and make a new release to crates.io :-)

**Comment by duesee** [comment]
Here we go https://crates.io/crates/abnf 🎉

**Pull request merged**